core/store/src/trie/trie_storage.rs

```diff
@@ -119,7 +119,7 @@ impl TrieStorage for TrieMemoryPartialStorage {
 
 /// Maximum number of cache entries.
 #[cfg(not(feature = "no_cache"))]
-const TRIE_MAX_CACHE_SIZE: usize = 10000;
+const TRIE_MAX_CACHE_SIZE: usize = 50000;
```

Comment:
please add a small comment describing how this value was chosen (it is fine to say 'randomly' or 'best guess') - and what is the impact on the resources used -for example RAM.